### PR TITLE
feat: market-data slice improvements

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR removes `findByAssetIds` and `findPriceHistoryByAssetIds` endpoints in profit of `findByAssetId` and `findPriceHistoryByAssetId`.

This ensures:

- better ergonomics (i.e many many places we were passing an array of single AssetId)
- most importantly, better performance. See https://github.com/shapeshift/web/issues/7599, which, while it *is* an extreme case, would've been somehow mitigated to *some* extent by leveraging the caching of previous market-data RTK queries. 
  The reason is RTK query args are shallow compared, and having the same AssetId in different `assetIds[]` args between runs means market-data will be refetched. By fetching one vs. many, we avoid this and ensure caching works as it should.

## Issue (if applicable)

N/A, though relates to the closed https://github.com/shapeshift/web/issues/7599

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low, but effectively medium given the surface area. See testing steps.

## Testing

Ensure market-data is happy across the app:

- Dashboard
- Assets Page
- Asset Page
- Account Asset Page
- Swapper
- Fiat ramps
- DeFi, Lending, LP, RFOX

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

https://jam.dev/c/464026f7-308e-4061-b94a-b1cd54a609f3

### CODE REVIEW

File: src/context/AppProvider/AppContext.tsx [hunk 1]
```diff
@@ -246,11 +246,15 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useQuery({
     queryKey: ['marketData', {}],
     queryFn: async () => {
-      await dispatch(
-        marketApi.endpoints.findByAssetIds.initiate(portfolioAssetIds, {
-          // Since we use react-query as a polling wrapper, every initiate call *is* a force refetch here
-          forceRefetch: true,
-        }),
+      await Promise.all(
```
Comment (gomesalexandre) [inline src/context/AppProvider/AppContext.tsx:249]: `useQueries`

**Comment by 0xean** [comment]
@gomesalexandre - is there a groomed ticket associated with this work?

File: src/context/AppProvider/AppContext.tsx [hunk 2]
```diff
@@ -243,31 +243,33 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
...
-  useQuery({
-    queryKey: ['marketData', {}],
-    queryFn: async () => {
-      await dispatch(
-        marketApi.endpoints.findByAssetIds.initiate(portfolioAssetIds, {
-          // Since we use react-query as a polling wrapper, every initiate call *is* a force refetch here
-          forceRefetch: true,
-        }),
-      )
+  useQueries({
```
Comment (NeOMakinG) [inline src/context/AppProvider/AppContext.tsx:246]: 🧠

File: src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts [hunk 3]
```diff
@@ -16,47 +16,53 @@ export const useFetchPriceHistories = (assetIds: AssetId[], timeframe: HistoryTi
   const symbol = useAppSelector(selectSelectedCurrency)
 
   useEffect(() => {
-    dispatch(findPriceHistoryByAssetIds.initiate({ assetIds, timeframe }))
+    assetIds.forEach(assetId => {
+      dispatch(marketApi.endpoints.findPriceHistoryByAssetId.initiate({ assetId, timeframe }))
+    })
```
Comment (NeOMakinG) [inline src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts:21]: q: I don't know if it's managed under the hood but by reading this little piece of code without knowing the underlying context, I feel like we are rerunning all the price history fetching for every assets if a small piece inside `assetIds` changes

**Review by NeOMakinG — COMMENTED** [review]
Looks good so far!

**Review by NeOMakinG — APPROVED** [review]
https://jam.dev/c/1b765ee4-5c79-44e1-9107-c64dd3ffb685

Couldn't find any regression, nice job!

File: src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts (hunk 3 above)
Comment (gomesalexandre) [reply src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts:21]: We don't `forceRefetch` here so we we won't refetch, only fetch fresh ones here

**Pull request merged**